src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,99 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private Object[] elements;
```

Comment:
There is should be parametrised type

Reply:
If element's type is parameterized, then it needs to cast 'elements' array inside a constructor anyway. Is that a major mistake to use 'Object' type for an array?

Reply:
Not major and this isn't a mistake. It's like another option of implementation.